scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/Rewrite.scala

```diff
@@ -66,7 +67,11 @@ case class RewriteCtx(
 }
 
 abstract class Rewrite {
-  def rewrite(implicit ctx: RewriteCtx): Unit
+  def create(implicit ctx: RewriteCtx): RewriteSession
+}
+
+abstract class RewriteSession(implicit ctx: RewriteCtx) {
```

Comment:
What is purpose to make `ctx` implicit?

Reply:
less code to write, don't need to pass it around. perhaps this is also the result of our general rule that it's a good idea to use implicits for unique instances of a less common type. if you prefer, can make it explicit easily, please let me know.

Reply:
I prefer explicit constructors but it doesn't really matter